pkg/operator/target_config_reconciler.go

```diff
@@ -551,6 +595,28 @@ func (c *TargetConfigReconciler) manageServiceMonitor(ctx context.Context, owner
 		ownerReference,
 	})
 
+	service := resourceread.ReadServiceV1OrDie(bindata.MustAsset("assets/lws-controller-generated/v1_service_lws-controller-manager-metrics-service.yaml"))
+	endpoints, found, err := unstructured.NestedSlice(required.Object, "spec", "endpoints")
+	if !found || len(endpoints) == 0 || err != nil {
+		return false, fmt.Errorf("%v: .spec.endpoints not found: %v", required.GetName(), err)
+	}
+
+	serverName, found, err := unstructured.NestedString(endpoints[0].(map[string]any), "tlsConfig", "serverName")
+	if !found || err != nil {
+		return false, fmt.Errorf("%v: .spec.endpoints.0.tlsConfig.serverName not found: %v", required.GetName(), err)
+	}
+	serverName = strings.Replace(serverName, "SERVICE_NAME", service.Name, 1)
+	serverName = strings.Replace(serverName, "SERVICE_NAMESPACE", c.namespace, 1)
+	err = unstructured.SetNestedField(endpoints[0].(map[string]any), serverName, "tlsConfig", "serverName")
+	if err != nil {
+		return false, err
+	}
+
+	err = unstructured.SetNestedSlice(required.Object, endpoints, "spec", "endpoints")
+	if err != nil {
+		return false, err
+	}
+
```

Comment:
IMO, it would be better to also use the prometheus secret that we already have instead of the operand cert: 

https://github.com/openshift/jobset-operator/blob/61c48a1547d2b12f19dee83b5c4b4d71caba6372/pkg/operator/target_config_reconciler.go#L372

Reply:
So I added your ServiceMonitor implementation and adapted it to my PR.